src/extensions/layout-selector/index.js

```diff
@@ -92,6 +93,28 @@ class LayoutSelector extends Component {
 		this.useTemplateLayout = this.useTemplateLayout.bind( this );
 		this.useEmptyTemplateLayout = this.useEmptyTemplateLayout.bind( this );
 		this.renderContent = this.renderContent.bind( this );
+		this.detectImageBlocks = this.detectImageBlocks.bind( this );
+		this.uploadExternalImages = this.uploadExternalImages.bind( this );
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.clientIds.length === 0 && this.props.clientIds.length !== 0 ) {
```

Comment:
@jrtashjian This is a big If statement. I would find the reverse condition and return early instead if possible.

Reply:
Fixed in 4d05075